devito/function.py

```diff
@@ -955,17 +1019,40 @@ def interpolate(self, expr, offset=0, u_t=None, p_t=None, cummulative=False):
                       for v in variables]
             idx_subs += [OrderedDict(v_subs)]
         # Substitute coordinate base symbols into the coefficients
-        subs = OrderedDict(zip(self.point_symbols, self.coordinate_bases))
+        subs = OrderedDict(zip(self.point_symbols, self.coordinate_bases()))
         rhs = sum([expr.subs(vsub) * b.subs(subs)
                    for b, vsub in zip(self.coefficients, idx_subs)])
         # Apply optional time symbol substitutions to lhs of assignment
-        lhs = self if p_t is None else self.subs(self.indices[0], p_t)
+        lhs = lhs if p_t is None else lhs.subs(self.indices[0], p_t)
 
         rhs = rhs + lhs if cummulative is True else rhs
 
         return [Eq(lhs, rhs)]
 
     def inject(self, field, expr, offset=0, u_t=None, p_t=None):
+        """
+        Symbol for injection of an expression onto a grid
+
+        :param field: The grid field into which we inject.
+        :param expr: The expression or list of fd expression to inject.
+        :param offset: Additional offset from the boundary for
+                       absorbing boundary conditions.
+        :param u_t: (Optional) time index to use for indexing into `field`.
```

Comment:
I don't remember if I had written it already -- if yes, apologies for saying it again -- but these two parameters ``u_t`` and ``p_t`` are only meaningful in a time object

Reply:
Done